ietf/meeting/views.py

```diff
@@ -2516,6 +2527,59 @@ def add_session_drafts(request, session_id, num):
                     'form': form,
                   })
 
+def bluesheet_data(session):
+    def affiliation(meeting, person):
+        # from OidcExtraScopeClaims.scope_registration()
+        email_list = person.email_set.values_list('address')
+        q = Q(person=person, meeting=meeting) | Q(email__in=email_list, meeting=meeting)
+        regs = MeetingRegistration.objects.filter(q).distinct()
+        return ([reg.affiliation for reg in regs if reg.affiliation] or [''])[0]
+
+    attendance = Attended.objects.filter(session=session)
+    meeting = session.meeting
+    return [{'name':attended.person.name, 'affiliation':affiliation(meeting, attended.person)} for attended in attendance]
+
+def session_attendance(request, session_id, num):
```

Comment:
This needs to not allow someone to say "I attended" before they attended. It should probably not let people in at all until at least one Attended record exists for a session. That's counting on a side-effect since the only other way such a record would be created is MeetEcho's upload right now, but we should take advantage of that for now - otherwise we have to define when the view should open based on some calculation of "the session is over".

Reply:
Using that "side effect" has the nice property that it mostly prevents impatient people from self-attesting their attendance before MeetEcho uploads their records. I.e., I think it's pretty close to the right signal to open this mechanism up.